lib/units/storage/amazons3.js

```diff
@@ -0,0 +1,197 @@
+var http = require('http')
+var util = require('util')
+var path = require('path')
+
+var express = require('express')
+var validator = require('express-validator')
+var bodyParser = require('body-parser')
+var formidable = require('formidable')
+var Promise = require('bluebird')
+var uuid = require('node-uuid')
+var fs = require('fs')
+
+var lifecycle = require('../../util/lifecycle')
+var logger = require('../../util/logger')
+var requtil = require('../../util/requtil')
+
+var AWS = require('aws-sdk')
+
+module.exports = function(options) {
+  var log = logger.createLogger('storage:s3')
+  , app = express()
+  , server = http.createServer(app)
+  , credentials = new AWS.SharedIniFileCredentials({
+      profile: options.profile
+    })
+  , bucket = options.Bucket
+  , s3
+
+  // TODO: remove later
+  log.debug(options)
+
+  AWS.config.credentials = credentials;
+  s3 = new AWS.S3(options)
```

Comment:
I would just put the `var` here and remove it from the top of the function body. I know we're doing it like that in many places, but it's usually only in small functions where you can see the whole source easily. Here it looks too much like a global variable.